src/item.cpp

```diff
@@ -4348,10 +4348,24 @@ bool item::is_power_armor() const
 
 int item::get_encumber( const Character &p ) const
 {
+    units::volume char_storage( 0_ml );
     units::volume contents_volume( 0_ml );
+
     for( const item &e : contents ) {
         contents_volume += e.volume();
     }
+
+    for( const item &e : p.worn ) {
+        char_storage += e.get_storage();
+    }
+    if( char_storage != 0_ml ) {
+        const islot_armor *t = find_armor_data();
+        if( t != nullptr ) {
+            contents_volume += units::from_milliliter( t->storage.value() * p.inv.volume().value() /
+                               char_storage.value() );
+        }
+    }
```

Comment:
This new chunk of code should probably only run when `t->max_encumber != 0`, otherwise it will affect the encumbrance for e.g. waterskins.